mne/forward/_field_interpolation.py

```diff
@@ -87,6 +107,84 @@ def _compute_mapping_matrix(fmd, info):
     return mapping_mat
 
 
+def compute_virtual_evoked(evoked, from_type='mag', to_type='grad',
+                           copy=True, baseline=None, mode='fast', n_jobs=1):
+    """Compute virtual evoked using interpolated fields in mag/grad channels.
+
+    Parameters
+    ----------
+    evoked : instance of mne.Evoked
+        The evoked object.
+    from_type : str
+        The original channel type. It can be 'mag' or 'grad'.
+    to_type : str
+        The destination channel type. It can be 'mag' or 'grad'.
+    copy : bool
+        If False evoked is modified in place.
+    n_jobs : int
+        Number of jobs to run in parallel.
+
+    Returns
+    -------
+    evoked : instance of mne.Evoked
+        The transformed evoked object containing only virtual channels.
+    """
+    if from_type not in ['mag', 'grad']:
+        raise ValueError('from_type must be "mag" or "grad", not "%s"'
+                         % from_type)
+    if to_type not in ['mag', 'grad']:
+        raise ValueError('to_type must be "mag" or "grad", not "%s"'
+                         % to_type)
+    if from_type == to_type:
+        raise ValueError('from_type and to_type cannot be the same.')
+
+    # pick the original and destination channels
+    pick_from = pick_types(evoked.info, meg=from_type, eeg=False,
+                           ref_meg=False)
+    pick_to = pick_types(evoked.info, meg=to_type, eeg=False,
+                         ref_meg=False)
+
+    info_from = pick_info(evoked.info, pick_from, copy=True)
+    info_to = pick_info(evoked.info, pick_to, copy=True)
+
+    # no need to apply trans because both from and to coils are in device
+    # coordinates
+    coils_from = _create_coils(info_from['chs'], FIFF.FWD_COIL_ACCURACY_NORMAL,
+                               info_from['dev_head_t'], 'meg')
+    coils_to = _create_coils(info_to['chs'], FIFF.FWD_COIL_ACCURACY_NORMAL,
+                             info_to['dev_head_t'], 'meg')
+    miss = 1e-4  # Smoothing criterion for MEG
+    ch_type = 'meg'
+    #
+    # Step 2. Calculate the dot products
+    #
+    my_origin, int_rad, noise, lut_fun, n_fact = _setup_dots(mode, coils_from,
+                                                             ch_type)
+    logger.info('Computing dot products for %i coils...' % (len(coils_from)))
+    self_dots = _do_self_dots(int_rad, False, coils_from, my_origin, ch_type,
+                              lut_fun, n_fact, n_jobs)
+    logger.info('Computing cross products for coils %i x %i coils...'
+                % (len(coils_from), len(coils_to)))
+    cross_dots = _do_cross_dots(int_rad, False, coils_from, coils_to,
+                                my_origin, ch_type, lut_fun, n_fact).T
+
+    ch_names = [c['ch_name'] for c in info_from['chs']]
+    fmd = dict(kind=ch_type, ch_names=ch_names,
+               origin=my_origin, noise=noise, self_dots=self_dots,
+               surface_dots=cross_dots, int_rad=int_rad, miss=miss)
```

Comment:
How much of this code is shared with the make_field_map code? It would be good to stay as DRY as possible and unify where we can.